services/core-api/app/api/now_applications/resources/now_application_delay_resource.py

```diff
@@ -65,6 +74,11 @@ def put(self, now_application_guid, now_application_delay_guid):
         if not now_app:
             raise NotFound('Notice of Work Application not found')
 
+        # change NoW status back to previous state
```

Comment:
Does this not get triggered every time this method is called?

Reply:
yes - this method is only called when setting an end_date on the delay, so If the application record exists, revert the status back. There are no other checks needed.